modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -0,0 +1,616 @@
+
+        // Prepare XLinkOut nodes for each output object in graph
+        for (size_t i = 0; i < outs_data.size(); ++i) {
+            auto xout = m_pipeline->create<dai::node::XLinkOut>();
+            std::string xout_name = "xout" + std::to_string(i);
+            m_out_queue_names.push_back(xout_name);
+            xout->setStreamName(xout_name);
+            m_xlink_outputs.push_back(xout);
+        }
+
+        // Create OAK node for each node in this backend
+        for (const auto& nh : nodes) {
+            if (m_gm.metadata(nh).get<NodeType>().t == NodeType::OP) {
+                const auto& op = m_gm.metadata(nh).get<Op>();
+                const auto &u = ConstOAKGraph(m_g).metadata(nh).get<OAKComponent>();
+                // pass kernel input args and compile args to prepare OAK node and
+                // store it to link later
+                m_oak_nodes[nh] = nullptr;
+                m_oak_node_inputs[nh] = {};
+                m_oak_node_outputs[nh] = {};
+
+                std::vector<cv::GArg> in_ctx_args;
+                in_ctx_args.reserve(op.args.size());
+                for (auto &op_arg : op.args) in_ctx_args.push_back(packArg(op_arg,
+                                                                           m_oak_node_inputs[nh],
+                                                                           m_oak_wrapped_args));
+                for (auto &&op_out : op.outs) outArg(op_out, m_oak_node_outputs[nh]);
+                GAPI_Assert(!m_oak_node_inputs[nh].empty());
+                GAPI_Assert(!m_oak_node_outputs[nh].empty());
+
+                GOAKContext ctx(m_pipeline, m_camera_wh, in_ctx_args, m_oak_node_outputs[nh]);
+                u.k.m_f(ctx, m_oak_nodes[nh], m_in_queues);
+                GAPI_Assert(m_oak_nodes[nh] != nullptr);
+            }
+        }
+
+        for (const auto& el : m_oak_node_inputs) {
+            for (const auto& in : el.second) {
+                GAPI_Assert(in != nullptr);
+            }
+        }
+
+        for (const auto& el : m_oak_node_outputs) {
+            for (const auto& out : el.second) {
+                GAPI_Assert(out != nullptr);
+            }
+        }
+
+        // Properly link all nodes
+        // 1. Link input nodes to camera
+        for (const auto& d : ins_data)
+        {
+            for (const auto& nh : nodes)
+            {
+                if (m_gm.metadata(nh).contains<cv::gimpl::Op>())
+                {
+                    for (const auto& indata : nh.get()->inNodes()) {
+                        auto rc = m_gm.metadata(indata).get<cv::gimpl::Data>().rc;
+                        if (rc == d.rc)
+                        {
+                            GAPI_Assert(m_oak_nodes.find(nh) != m_oak_nodes.end());
+                            GAPI_Assert(m_oak_node_inputs[nh].size() == 1);
+                            // FIXME: covert other camera outputs
+                            m_camera_input->video.link(*(m_oak_node_inputs[nh][0]));
```

Comment:
a-ha moment here - so given that you do `*`, you don't need `Input/OutputPtr*`.

Reply:
I just checked `operator=` is unavailable for `Input/Output`

Reply:
So, we can't use `*in = edgeDetector->inputImage;` for `GOAKContext::InputPtr in`. It seems we can only work with `**`. Please, correct me if I'm wrong